move common hardware attributes and metrics to registry

trisch-me opened the pull request:
This is the first PR out of many for hardware metrics.

The first PR is about generic attributes used by all hardware metrics. I decided to split it into multiple PRs to make review easier because it's easy to miss something with a lot of changed files (I have started doing it with 5 metrics and lost myself track on things :) ). 

I delete portions of `docs/system/hardware-metrics.md` file so that reviewers could see what has been changed now. This file will be deleted later as soon as it will be ported completely.

Hardware wasn't moved to the yaml schema, so I'm doing here both introducing it to the registry and creating appropriate yaml files for each hardware metric

I need this PR merged before I will create other PRs (half of them is ready but based on this PR)

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

lmolkova commented on model/metrics/hardware/common.yaml:
```diff
@@ -0,0 +1,59 @@
+    type: metric
+    metric_name: hw.energy
+    stability: experimental
+    brief: "Energy consumed by the component, in joules"
```
```suggestion
    brief: "Energy consumed by the component"
```

lmolkova commented on model/metrics/hardware/common.yaml:
```diff
@@ -0,0 +1,59 @@
+    unit: "{errors}"
+    extends: metric.hardware.attributes
+    attributes:
+      - ref: hw.error.type
```
```suggestion
      - ref: error.type
```

trisch-me replied:
I have answered in another similar comment

lmolkova replied:
Since there was no defined attribute for `hw.error.type`, my strong preference would be not to introduce one and instead use generic `error.type`.

Otherwise, we'd end up with another attribute the should be immediately deprecated.

lmolkova commented on model/metrics/hardware/common.yaml:
```diff
@@ -0,0 +1,59 @@
+    type: metric
+    metric_name: hw.power
+    stability: experimental
+    brief: "Instantaneous power consumed by the component, in Watts"
```
```suggestion
    brief: "Instantaneous power consumed by the component"
```

lmolkova commented on model/metrics/hardware/common.yaml:
```diff
@@ -0,0 +1,59 @@
+
+  - id: metric.hardware.energy
+    type: metric
+    metric_name: hw.energy
```
```suggestion
    metric_name: hw.energy.consumption
```
?

trisch-me replied:
I don't change in this PR anything, I do only refactoring (in cases where it's possible), if we would like to change names and introduce a breaking change - this should be done later for simplicity

bertysentry replied:
@lmolkova This can be discussed in a separate issue. The `.consumption` suffix is not necessary. Also, this counter may also cover energy production (batteries, for example), in which case we will use an attribute to indicate the *direction* (`direction="out|in"`).

lmolkova replied:
created https://github.com/open-telemetry/semantic-conventions/issues/1071 to follow up

lmolkova commented on model/metrics/hardware/common.yaml:
```diff
@@ -0,0 +1,59 @@
+    type: metric
+    metric_name: hw.power
+    stability: experimental
+    brief: "Instantaneous power consumed by the component, in Watts"
```
is it derivable from `hw.energy` ? why is this one necessary?

bertysentry replied:
You're right! It has been decided to specify both `hw.power` and `hw.energy` because:

* `hw.energy` (counter in Joules) is preferred because it provides more accurate aggregation (just like network traffic or CPU time utilization is better calculated with cumulative counters instead of averaging rates)
* Unfortunately, most systems are only capable of reporting their instantaneous electrical power consumption in Watts. To make sure existing systems all use the same metric name for electrical power, we specified `hw.power` in the semantic conventions.

Again: It can be discussed in a separate issue.

lmolkova commented on model/registry/hardware.yaml:
```diff
@@ -0,0 +1,113 @@
+        brief: >
+          An easily-recognizable name for the hardware component
+        examples: ['eth0']
+      - id: parent
```
```suggestion
      - id: parent.id
```

trisch-me replied:
This will introduce a breaking change but probably we can't go without one as `parent` is very common

bertysentry replied:
Personal opinion: Let's not put namespaces everywhere. Will there ever be `parent.name`, or `parent.parent`?

`parent` is indeed very generic. If it belongs to the hardware semantic conventions only, it could be later changed to `hw.parent`. 

However, I think this attribute could be beneficial to other semconv, like process, system disks and filesystems, databases, etc. Maybe it could be specified as a general attribute, applicable everywhere.

This will be discussed in a separate issue.

trisch-me replied:
@bertysentry there is a existing `process.parent` fields from ECS. But I suspect parent could be used also somewhere else. So I am also for introducing a separate namespace for it. But I don't want to do this in this PR. I want this PR to be as plain as possible - only move things out to the yaml files and the registry without functional changes

bertysentry replied:
Understood, thank you @trisch-me!

lmolkova commented on model/registry/hardware.yaml:
```diff
@@ -0,0 +1,113 @@
+        type:
+          allow_custom_values: true
+          members:
+            - id: battery.yaml
```
```suggestion
            - id: battery
```

lmolkova commented on model/registry/hardware.yaml:
```diff
@@ -0,0 +1,113 @@
+        stability: experimental
+        brief: >
+          Type of the component
+      - id: error.type
```
there is no need to declare `hw.error.type`, we should reuse the existing one

trisch-me replied:
this error type has specific enum value (will be updated in the future PRs). I can't redefine enum values from existing one in current build system.

bertysentry replied:
We need extensible enums, because the semantic conventions cannot cover all possible values of error types.

trisch-me replied:
@bertysentry we have discussed extensible enums already, maybe they will be there in the near future 
cc @jsuereth @lquerel 
But as for now I will just redefine enum with all possible options as soon as I will move them over

bertysentry replied:
Alright, thank you!

lmolkova replied:
error.type is intended to be common attribute describing an error, individual semconv can define additional domain specifc attributes, or specify what `error.type` means in their context. 

Let's reuse the generic one as the first step.

bertysentry replied:
@lmolkova The intent of this PR is just to migrate from Markdown to YAML. Semantic changes must be brought in a separate PR.

trisch-me replied:
@lmolkova I would be ok to use existing `error.type` but I don't see how I can redefine enum unless I redefine the global one.

I would need to make `error.type` an enum of fields such as for example `zero_buffer_credit`, `crc`, `corrected` etc depending on the type of hardware. There is no reusable enums currently. So the only thing I could do here would be update comments and put examples there. 

@bertysentry is it ok from your perspective if type of error is not defined as enum but could have any value? We do not restrict enums currently anyway but sometimes it is nice to have predefined set of values. Here I can put set of values only to examples session

lmolkova commented on model/registry/hardware.yaml:
```diff
@@ -0,0 +1,113 @@
+              value: 'tape_drive'
+              brief: 'Tape drive'
+              stability: experimental
+            - id: temperature
```
hw.type = temperature? What it be used for? cooling system? temperature sensor?

trisch-me replied:
it is defined as temperature sensor in the hardware-metrics. The same goes for voltage

bertysentry replied:
@lmolkova It's a matter of opinion: `temperature` is not confusing to me. 

Appending `_sensor` doesn't provide additional meaning, as most metrics in hardware are reported by "sensors" anyway: power status, memory presence, temperature, fan speed, etc. Why add `_sensor` for temperatures and voltages, but not for fan speed or power supplies? Let's keep this simple and consistent.

lmolkova replied:
`hw.type = fan`  makes sense, fan is a hardware component, same with `power_supply`.
`temperature` is not a component.

bertysentry replied:
Okay @lmolkova now I understand why it sounds confusing. Actually, `hw.type` is not to be read as "hardware type", but as the "type" attribute used by various metrics (notably `hw.status`), which happens to be in the `hw.` namespace.

`hw.status` can be used for disks, network cards, memory modules, etc. and temperatures, voltages, etc. `hw.status{hw.type="temperature"}` will report whether temperature is okay or not, and not whether the temperature sensor is operating properly. `hw.status{hw.type="temperature_sensor"}` would actually be very confusing to me, as I'd interpret it as whether the sensor is working fine or not.

I'm not sure however if we can clarify this in the description of the attribute. I still hope my explanation above makes it clearer! 😅

lmolkova replied:
I see, thanks for the explanation!

I agree that `temperature_sensor` would not make sense either then.

I guess that's another big thing to revisit in the https://github.com/open-telemetry/semantic-conventions/issues/1071 and should not block this PR.

So for the future
- `hw.type` name could be more precise. But also wouldn't it be clearer to report temperature as a separate metric (attribute on spans) and let someone decided if it's good/bad at a query time?  `hw.type=temperature, hw.status=failed` does not quite make sense either
- we might need to break this enum down into a component type (cpu|fan|etc) & component state|indicator|etc?. E.g. if I want to report CPU temperature, how do I do it today? `name="AMD Rizen 42",type=temperature,status=ok` ?

For this PR:

can we clarify what's `hw.type` is in the `hw.type` note?  E.g.

> Describes the category of hardware component for which `hw.state` is reported. For example, `hw.type=temperature` along with `hw.state=degraded` describes that the temperature of the hardware component has degraded.

lmolkova commented on model/registry/hardware.yaml:
```diff
@@ -0,0 +1,113 @@
+    prefix: hw
+    type: attribute_group
+    brief: >
+      Attributes for hardware.
```
It sounds the goal is to describe hardware component rather than a generic hardware, so I'd prefer to have `hw.component` or `hardware.component` as a namespace

trisch-me replied:
Can we change the whole namespace? It will go from current `hw.*` to `hw.XX.*` which means breaking changes for **all** fields

bertysentry replied:
I strongly disagree with this. In hardware, we're specifying metrics that apply to different levels: `hw.energy` may apply to a processor, a server, a UPS or even a data center. Same for temperature. Hardware does include components, but it's not just that, and it doesn't need to be namespaced as there are no collisions here.

lmolkova replied:
well, the `hw.id` seems to be better called `hw.component.id`.  While some things can stay in the `hw` root namespace, some might need more love to become descriptive

lmolkova commented on model/registry/hardware.yaml:
```diff
@@ -0,0 +1,113 @@
+              value: 'temperature'
+              brief: 'Temperature'
+              stability: experimental
+            - id: voltage
```
same here  - `voltage` seems confusing as a type

lmolkova replied:
so the suggestion is to use `voltage_sensor`

bertysentry replied:
@lmolkova See my previous comment for temperature.

lmolkova commented on docs/system/hardware-metrics.md:
```diff
@@ -12,9 +12,7 @@ when creating instruments not explicitly defined in the specification.
 
 <!-- toc -->
 
-- [Common hardware attributes](#common-hardware-attributes)
```
it should still be relevant for other metrics defined in this document

trisch-me replied:
This file will be removed. I have deleted this portion only for visibility on what is moved out

lmolkova replied:
In the meantime, while it is not removed, could you please add a link to the section in the new file so that someone looking at this file will know that there are some common attributes?

lmolkova commented on docs/system/hardware-metrics.md:
```diff
@@ -44,43 +42,8 @@ when creating instruments not explicitly defined in the specification.
-
 ## Metric Instruments
 
-### `hw.` - Common hardware metrics
```
Can we keep HW metrics in the same file? 
If we're splitting them in 2, we should at least have cross-references and some explanation why they are different

/cc @open-telemetry/semconv-system-approvers

trisch-me replied:
Sorry, I don't understand your question? In which file?

lmolkova replied:
Currently hardware metrics are spread across multiple files. I understand the intention to move all of them, but we should keep things consistent in the mean time.

Since you're removing some metrics from this file, please at least add a link to the new doc from this one.

lmolkova replied:
Please add a link to the common metrics section of the new document, so people would see it among other metrics (and not just in the note above)

trisch-me replied:
Updated

lmolkova commented on docs/hardware/common.md:
```diff
@@ -0,0 +1,223 @@
+
+**Status**: [Experimental][DocumentStatus]
+
+All metrics in `hw.` instruments should be attached to a [Host Resource](/docs/resource/host.md)
```
Is is a responsibility of instrumentation author or the end user?

trisch-me replied:
This is refactoring (copy-paste) from original file.

bertysentry replied:
Responsibility is on the instrumentation developer.

bertysentry replied:
@lmolkova I'm sorry, but there must be a misunderstanding here. Let's say I'm HPE developing an OpenTelemetry instrumentation for my ProLiant servers, I will use an Otel SDK to push metrics that I captured through specific hardware busses, and while developing this software layer, I will need to create a Resource to attach my metrics to, so that both Resources and Metrics are sent over OTLP.

It's simply a matter of making sure all metrics have the proper attributes to attach them to a host.

Without these, it's practically impossible to use these metrics in real life.

bertysentry replied:
@lmolkova I understand. 

For my own knowledge, could you please clarify what you call "instrumentation"?

trisch-me replied:
Should I rephrase this sentence then such as 
```suggestion
Application developers should attach all the metrics in the `hw` namespace to a `Host Resource](/docs/resource/host.md)."
```

lmolkova commented on docs/hardware/common.md:
```diff
@@ -0,0 +1,223 @@
+
+These common `hw.` metrics include the below attributes to describe the
+monitored component:
+<!-- semconv metric.hardware.attributes(full) -->
```
we have another attribute table defined above, which of them should be applied to metrics?

trisch-me replied:
In original md file here was defined only `type` 
common attributes extend the table above therefore when you render it it has both previous attributes and `type`. Do you know how to skip it or should we skip it?

lmolkova replied:
Since each metric in this file defines extra attributes in addition to common ones, I suggest removing common attributes and just generating full attribute tables for each metric.

Otherwise it becomes hard to understand from the reader perspective which attribute apply to a specific metric without scrolling back and forth

bertysentry replied:
I agree with @lmolkova's suggestion here.

lmolkova reviewed (commented):
Please also add a changelog entry.

trask commented:
cc @bertysentry

trisch-me commented:
@lmolkova I'm confused by your comments. I haven't created anything new. I have extracted hardware from one md file into yaml. As I stated in description I will do this in batches because there will be open questions and to make review easier, so I have extracted only first portion

bertysentry commented:
@trisch-me Thank you so much for taking the time to move the hardware semconv to YAML! (I was supposed to do it but couldn't find the time... and I was afraid the tooling wouldn't work well on some of the specificities of the hardware metrics, with common attributes, etc.).

Let me know if you need any help with that!

If you want to see these metrics in real life (to get a better understanding of their purpose and use), you can check these Grafana dashboards:

* [https://hws-demo.sentrysoftware.com/d/SHFBpSH7z/hardware-sentry---site?var-hw_type=All&var-site=Sentry-Ottawa&var-host_name=idrac-r640-01.internal.sentrysoftware.net&var-agent=demo-hws-02.internal.sentrysoftware.net&orgId=1&refresh=30s](Aggregated view of several systems)
* [https://hws-demo.sentrysoftware.com/d/c2K8skm4k/hardware-sentry---host?orgId=1&var-site=Sentry-Paris&var-host_name=idrac-r640-01.internal.sentrysoftware.net&var-hw_type=All&var-agent=demo-hws-02.internal.sentrysoftware.net](Detailed view of 1 system)

Thanks!

bertysentry commented:
@trisch-me I confirm we met in Paris! It's funny when you realize usernames in GitHub are actual people! 😅 

I'll be happy to review any PR on this topic!

trisch-me commented:
blocked by https://github.com/open-telemetry/semantic-conventions/pull/1049

lmolkova commented on model/metrics/hardware/common.yaml:
```diff
@@ -0,0 +1,59 @@
+    stability: experimental
+    brief: "Number of errors encountered by the component"
+    instrument: counter
+    unit: "{errors}"
```
```suggestion
    unit: "{error}"
```

bertysentry replied:
We used to use plural for non-standard units (in curly brackets). Has this changed?

lmolkova commented on .github/ISSUE_TEMPLATE/bug_report.yaml:
```diff
@@ -44,6 +44,7 @@ body:
         - area:gcp
         - area:gen-ai
         - area:graphql
+        - area:hardware
```
could we be consistent with the root namespace?
```suggestion
        - area:hw
```

(and in other places)

trisch-me replied:
hw as abbreviation seems strange to me in registry so I prefer hardware, but I haven't changed the namespace.
@bertysentry do you think we should/can change common namespace for hardware to be indeed `hardware` and not `hw`?

trisch-me replied:
@bertysentry could you please check this comment?

bertysentry replied:
I'm not quite sure. I don't think the `area` field is supposed to match the namespace (see `gen-ai` which is not namespace for generative AI metrics).

So I think it's okay to keep `area:hardware` value, as it's just one of the possible options in the issue template.

lmolkova commented:
> @lmolkova I'm confused by your comments. I haven't created anything new. I have extracted hardware from one md file into yaml. As I stated in description I will do this in batches because there will be open questions and to make review easier, so I have extracted only first portion

I created #1071 to keep track of HW naming issues. I still suggest to do a bare minimum changes to existing HW metrics so that we don't need to deprecated a bunch of things later (`hw.error.type`, and `hw.type` values).

bertysentry commented on docs/hardware/common.md:
```diff
@@ -158,11 +166,42 @@ This metric is [recommended][MetricRecommended].
+If a specific domain defines its own set of error identifiers (such as HTTP or gRPC status codes),
+it's RECOMMENDED to:
+
+* Use a domain-specific attribute
```
Given the description of the `error.type` attribute, I suggest that we enforce `error.type="hardware"` (or `error.type="hw"`?) and let the instrumentation use `hw.error.type` to specify the type of hardware error.

In issue #940, @eero-t suggested to add another attribute `hw.error.category`, to define even more precisely the type of error.

trisch-me replied:
later error.type would have precise specific values i.e. https://opentelemetry.io/docs/specs/semconv/system/hardware-metrics/#hwcpu---physical-processor-metrics or https://opentelemetry.io/docs/specs/semconv/system/hardware-metrics/#hwgpu---gpu-metrics so `error.type` will be defined for specific hardware component

eero-t replied:
> In issue #940, @eero-t suggested to add another attribute `hw.error.category`, to define even more precisely the type of error.

I changed my suggestion later to: https://github.com/open-telemetry/semantic-conventions/issues/940#issuecomment-2085109515
* `.severity`: correctable, uncorrectable (data loss), fatal (functionality loss)
* `.type`: type / category of the error; timeout, memory-parity-error, hw-programming-error etc
* `.location` / `.origin` / `.cause` / `.whatever`: e.g. http-endpoint, display (part of GPU with some memory), compute (GPU pipeline setting)

trisch-me replied:
This type of changes should be an additional PR after this one gets merged. So we don't mix refactoring (with minor changes) and major changes/new functionality

bertysentry commented on docs/hardware/common.md:
```diff
@@ -158,11 +166,42 @@ This metric is [recommended][MetricRecommended].
-| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| [`hw.error.type`](/docs/attributes-registry/hardware.md) | string | The type of error encountered by the component | `` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component [1] | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`error.type`](/docs/attributes-registry/error.md) | string | Describes a class of error the operation ended with. [2] | `timeout`; `java.net.UnknownHostException`; `server_certificate_invalid`; `500` | `Recommended` | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
```
The description is confusing in the context of hardware metrics.

trisch-me replied:
I can overwrite it here - what do you propose?

bertysentry commented on docs/attributes-registry/hardware.md:
```diff
@@ -10,14 +10,15 @@
+| ----------- | ------ | ------------------------------------------------------------------------------------------------------------- | ----------------------------------- | ---------------------------------------------------------------- |
+| `hw.id`     | string | An identifier for the hardware component, unique within the monitored host                                    | `win32battery_battery_testsysa33_1` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `hw.name`   | string | An easily-recognizable name for the hardware component                                                        | `eth0`                              | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `hw.parent` | string | Unique identifier of the parent component (typically the `id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0`                | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
```suggestion
| `hw.parent` | string | Unique identifier of the parent component (typically the `hw.id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0`                | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

bertysentry commented on docs/hardware/common.md:
```diff
@@ -97,10 +101,14 @@ This metric is [recommended][MetricRecommended].
-| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component [1] | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`hw.name`](/docs/attributes-registry/hardware.md) | string | An easily-recognizable name for the hardware component | `eth0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`hw.parent`](/docs/attributes-registry/hardware.md) | string | Unique identifier of the parent component (typically the `id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
```suggestion
| [`hw.parent`](/docs/attributes-registry/hardware.md) | string | Unique identifier of the parent component (typically the `hw.id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

bertysentry commented on docs/hardware/common.md:
```diff
@@ -158,11 +166,42 @@ This metric is [recommended][MetricRecommended].
+| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component [1] | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`error.type`](/docs/attributes-registry/error.md) | string | Describes a class of error the operation ended with. [2] | `timeout`; `java.net.UnknownHostException`; `server_certificate_invalid`; `500` | `Recommended` | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
 | [`hw.name`](/docs/attributes-registry/hardware.md) | string | An easily-recognizable name for the hardware component | `eth0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`hw.parent`](/docs/attributes-registry/hardware.md) | string | Unique identifier of the parent component (typically the `id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
```suggestion
| [`hw.parent`](/docs/attributes-registry/hardware.md) | string | Unique identifier of the parent component (typically the `hw.id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

bertysentry commented on docs/hardware/common.md:
```diff
@@ -224,10 +263,14 @@ This metric is [recommended][MetricRecommended].
-| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component [1] | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`hw.name`](/docs/attributes-registry/hardware.md) | string | An easily-recognizable name for the hardware component | `eth0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`hw.parent`](/docs/attributes-registry/hardware.md) | string | Unique identifier of the parent component (typically the `id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
```suggestion
| [`hw.parent`](/docs/attributes-registry/hardware.md) | string | Unique identifier of the parent component (typically the `hw.id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

bertysentry commented on docs/hardware/common.md:
```diff
@@ -290,10 +333,14 @@ This metric is [recommended][MetricRecommended].
-| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`hw.type`](/docs/attributes-registry/hardware.md) | string | Type of the component [1] | `battery`; `cpu`; `disk_controller` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`hw.name`](/docs/attributes-registry/hardware.md) | string | An easily-recognizable name for the hardware component | `eth0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | [`hw.parent`](/docs/attributes-registry/hardware.md) | string | Unique identifier of the parent component (typically the `id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
```suggestion
| [`hw.parent`](/docs/attributes-registry/hardware.md) | string | Unique identifier of the parent component (typically the `hw.id` attribute of the enclosure, or disk controller) | `dellStorage_perc_0` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

bertysentry commented:
We're not done yet!

trisch-me commented:
@lmolkova @bertysentry I have updated all the comments except the naming one. could you look once more

bertysentry commented:
Approved! Let's merge this so we can work the remaining issues!

lmolkova commented on model/registry/hardware.yaml:
```diff
@@ -0,0 +1,111 @@
+        examples: ['dellStorage_perc_0']
+      - id: type
+        type:
+          allow_custom_values: true
```
```suggestion
```

lmolkova commented on model/registry/hardware.yaml:
```diff
@@ -0,0 +1,111 @@
+          component has been reported as `degraded`.
+      - id: state
+        type:
+          allow_custom_values: true
```
```suggestion
```

lmolkova commented on docs/system/hardware-metrics.md:
```diff
@@ -44,43 +47,8 @@ when creating instruments not explicitly defined in the specification.
 > * SHOULD introduce a control mechanism to allow users to opt-in to the new
 >   conventions once the migration plan is finalized.
 
-## Common hardware attributes
```
could you please add link from this document to the common.md#common-attributes and mention that attributes are defined there. Attributes declared here  apply to all metrics in this file and with them removed, this file is not complete.

trisch-me replied:
Updated

lmolkova commented on model/hardware-common.yaml:
```diff
@@ -0,0 +1,12 @@
+  - id: hardware.attributes.common
+    type: attribute_group
+    stability: experimental
+    brief: 'Hardware common attributes'
```
```suggestion
    brief: 'Common hardware attributes'
```

lmolkova commented on docs/hardware/common.md:
```diff
@@ -0,0 +1,380 @@
+<!-- END AUTOGENERATED TEXT -->
+<!-- endsemconv -->
+
+<!-- semconv metric.hardware.energy(full) -->
```
```suggestion
<!-- semconv metric.hardware.energy -->
```

(and in other places)

lmolkova commented on model/metrics/hardware/common.yaml:
```diff
@@ -0,0 +1,61 @@
+      - ref: error.type
+        brief: "The type of error encountered by the component"
+        examples: ['uncorrected', 'zero_buffer_credit', 'crc', 'bad_sector']
+        requirement_level: recommended
```
```suggestion
        requirement_level:
          conditionally_required: if and only if an error has occured
        note: >
          The `error.type` SHOULD match the error code reported by the component,
          the canonical name of exception that occurred, or another low-cardinality error identifier.
          Instrumentations SHOULD document the list of errors they report.  
```

trisch-me replied:
I will update it as it makes more sense but I need to note that originally it was a `recommended`

lmolkova commented on model/metrics/hardware/common.yaml:
```diff
@@ -0,0 +1,61 @@
+    stability: experimental
+    brief: "Instantaneous power consumed by the component"
+    note: >
+      `hw.energy` is preferred
```
```suggestion
      It's RECOMMENDED to report `hw.energy` metric instead of `hw.power` when possible. 
```

lmolkova commented on docs/hardware/README.md:
```diff
@@ -0,0 +1,17 @@
+
+Semantic conventions for hardware are defined as following:
+
+* [Common Hardware Metrics](common.md): Semantic Conventions for hardware *common* metrics.
```
```suggestion
* [Common Hardware Metrics](common.md): Semantic Conventions for  *common* hardware metrics.
```

lmolkova commented on docs/hardware/common.md:
```diff
@@ -0,0 +1,380 @@
+linkTitle: Hardware Common Metrics
+--->
+
+# Semantic Conventions for Hardware Common Metrics
```
```suggestion
# Semantic Conventions for Common Hardware Metrics
```

lmolkova commented on docs/hardware/common.md:
```diff
@@ -0,0 +1,380 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: Hardware Common Metrics
```
```suggestion
linkTitle: Common Hardware Metrics
```

lmolkova commented on docs/system/hardware-metrics.md:
```diff
@@ -10,11 +10,14 @@ This document describes instruments and attributes for common hardware level
 when creating instruments not explicitly defined in the specification.
 
+This document is being converted to specific hardware metrics, parts of this document that have already been
+converted are now located in the [Hardware] (/docs/hardware/README.md) folder and are no longer present in this file.
```
```suggestion
converted are now located in the [Common Hardware Metrics](/docs/hardware/common.md) folder and are no longer present in this file.
```

trisch-me replied:
This is not correct, it is about all hardware metrics and not only common

lmolkova commented on docs/system/hardware-metrics.md:
```diff
@@ -10,11 +10,14 @@ This document describes instruments and attributes for common hardware level
+This document is being converted to specific hardware metrics, parts of this document that have already been
+converted are now located in the [Hardware] (/docs/hardware/README.md) folder and are no longer present in this file.
+
+Please note that this is an ongoing process and may take some time to complete.
```
do we have an issue that outlines what needs to happen? Can we add a todo comment here and add link to that issue?

trisch-me replied:
I have created a tracking issue for this https://github.com/open-telemetry/semantic-conventions/issues/1309
